### PR TITLE
Feature/matching service fetch categories

### PR DESCRIPTION
**Pull request by DonovanJJ**
# Changes
* Matching service fetches categories from question service for input validation
* Updated frontend matching form logic based on matching service api changes
* Standardised naming of `topics` into `categories`
* Added healthcheck for question service

### CODE REVIEW

File: frontend/src/features/matching/components/MatchingModal.tsx [hunk 1]
```diff
@@ -59,7 +59,7 @@ const MatchingModal: React.FC<MatchingModalProps> = ({
           },
           body: JSON.stringify({
             name: user?.username, // set up with user context later
-            topic: formData.topic,
```
Comment (ThatGirlJam) [inline frontend/src/features/matching/components/MatchingModal.tsx:62]: Thanks for changing the names from topic to category! 🙏

File: docker-compose.yml [hunk 2]
```diff
@@ -10,6 +10,12 @@ services:
       context: ./questionbank
     ports:
       - "8080:8080"
+    healthcheck:
```
Comment (ThatGirlJam) [inline docker-compose.yml:13]: Thanks for adding the healthchecks!

**Review by ThatGirlJam — APPROVED** [review]
LGTM! Thanks for the refactoring and additions 👍

**Review by seanpzk — APPROVED** [review]
LGTM, thanks for the prompt update to keep it working!

**Pull request merged**